packages/block-editor/src/components/block-list-appender/index.js

```diff
@@ -85,6 +85,22 @@ function BlockListAppender( {
 	tagName: TagName = 'div',
 } ) {
 	const appender = useAppender( rootClientId, renderAppender );
+	const isDragOver = useSelect(
+		( select ) => {
+			const {
+				getBlockInsertionPoint,
+				isBlockInsertionPointVisible,
+				getBlockCount,
+			} = select( blockEditorStore );
+			const insertionPoint = getBlockInsertionPoint();
+			return (
+				isBlockInsertionPointVisible() &&
+				rootClientId === insertionPoint?.rootClientId &&
+				getBlockCount( rootClientId ) === 0
+			);
```

Comment:
Given the insertion point can be visible not only for drag and drop operations, but also hover operations, should there be something here to check whether a drag is happening (e.g. `isDraggingBlocks`)?

I didn't see any issues, but it made me wonder 😄

Reply:
We only call `showInsertionPoint` for empty lists in `useBlockDropZone` so it will only show when dragging. But it's implicit and I think it makes sense to add the `isDragging` check if possible though!

Reply:
Here are some related articles about this:

- https://epicreact.dev/why-you-shouldnt-put-refs-in-a-dependency-array/
- https://stackoverflow.com/a/60476525

Reply:
Agree, I think it should be a matter of using that selector, so hopefully it's a straightforward change that makes the code more resilient to future changes.

Reply:
Turns out we currently don't have a hook to determine if we are in a dragging state. At least not without having to introduce a drop zone, which will intercept the behavior of this one. I can add a `useIsDragging` hook but I don't know where to put it. I think at this point we desperately need a minor refactor of DnD as it's making development unnecessarily hard 😅 .

Reply:
I added a comment for now. 👍